### DISCUSSION TITLE
Automatic upgrade migration from Wix 3.11(Microsoft.Tools.WindowsInstallerXml.Bootstrapper)custom UI code to Wix 4.0 wixtoolset.mba.core library

### DISCUSSION THREAD

**Discussion by shivajipatne** [wixtoolset/issues#7650] [Questions]
after upgrading from framework 4.6 to dotnet core 6.0  , my code is broken brutally , is that will support.

**Comment by robmen** [comment]
Converting BootstrapperApplications from v3 to v4 is involved. There are [a lot of breaking changes to consider](https://wixtoolset.org/docs/fourthree/faqs/#converting-bundles). At [FireGiant](https://www.firegiant.com/) we developed our BundleApplicationFramework to abstract away those complications for our customers. Updating to v4 was a bit of a process.

**Reply by Shivapatne** [reply]
can we get step by step process to do that work its help us lot?
